crml/sylo/src/e2ee/benchmarking.rs

```diff
@@ -0,0 +1,95 @@
+// Copyright 2019-2020 Centrality Investments Limited
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Sylo e2ee benchmarking.
+
+#![cfg(feature = "runtime-benchmarks")]
+
+use super::*;
+
+use frame_benchmarking::{account, benchmarks, whitelisted_caller};
+use frame_system::RawOrigin;
+use sp_runtime::traits::Hash;
+use sp_std::boxed::Box;
+use sp_std::{vec, vec::Vec};
+
+use crate::device::Module as SyloDevice;
+use crate::groups::{Group, Member, MemberRoles, Meta, Module as SyloGroups, Text};
+
+const SEED: u32 = 0;
+
+benchmarks! {
+	_{ }
+
+	register_device {
+		let sender: T::AccountId = whitelisted_caller();
+		let device_id: DeviceId = 11;
+
+		let text_tuple0 = (Text::from(*b"t0m0"), Text::from(*b"t0m1"));
+		let text_tuple1 = (Text::from(*b"t1m0"), Text::from(*b"t1m1"));
+		let meta = Meta::from([text_tuple0, text_tuple1]);
+
+		let admin = Member::<T::AccountId>::new(
+			sender.clone(),
+			Vec::<MemberRoles>::from([MemberRoles::Admin, MemberRoles::Member]),
+			Meta::new()
+		);
+
+		let member = Member::<T::AccountId>::new(
+			account("member", 0, SEED),
+			Vec::<MemberRoles>::from([MemberRoles::Member]),
+			Meta::new()
+		);
+
+		let group_id_0 = T::Hashing::hash(b"group0");
+		let group_0 = Group::<T::AccountId, T::Hash>::new(
+			group_id_0,
+			Vec::<Member<T::AccountId>>::from([admin.clone(), member.clone()]),
+			Vec::new(),
+			meta.clone(),
+		);
+		<SyloGroups<T>>::insert(&sender, &group_id_0, group_0);
+
+		let group_id_1 = T::Hashing::hash(b"group1");
+		let group_1 = Group::<T::AccountId, T::Hash>::new(
+			group_id_1,
+			Vec::<Member<T::AccountId>>::from([admin, member]),
+			Vec::new(),
+			meta,
+		);
+		<SyloGroups<T>>::insert(&sender, &group_id_1, group_1);
+
+		let pre_key_bundle0 = PreKeyBundle::from(*b"prekeybundle0");
+		let pre_key_bundle1 = PreKeyBundle::from(*b"prekeybundle1");
```

Comment:
Sylo registers something around 8-12 pkbs when this is called, could be worth adding a few more or adjusting the weight function as such

pkbs from the app are shas256 hashes / 64 bytes in length

Reply:
I fixed it to take the number of bundled pre-keys into account.